e2e/nomostest/nt.go

```diff
@@ -596,6 +593,40 @@ func (nt *NT) newPortForwarder(ns, deployment, port string, opts ...portforwarde
 	)
 }
 
+func (nt *NT) startPortForwarder(ns, deployment string, pf *portforwarder.PortForwarder) {
+	ctx, cancel := context.WithCancel(nt.Context)
+	eventCh := pf.Start(ctx)
+	doneCh := make(chan struct{})
+	readyCh := make(chan struct{})
+	nt.T.Cleanup(func() {
+		nt.T.Logf("killing port-forward %s/%s", ns, deployment)
+		cancel()
+		<-doneCh
+		nt.T.Logf("finished waiting for port-forward %s/%s to exit", ns, deployment)
+	})
+	go func() {
+		defer close(doneCh)
+		// first event should be Init. signal if it's ready
+		event := <-eventCh
+		if event.Type == portforwarder.InitEvent {
+			close(readyCh)
+		}
+		for event := range eventCh {
+			if event.Type == portforwarder.ErrorEvent {
+				nt.T.Error(event.Message)
+			}
+		}
+	}()
+	timeoutCtx, timeoutCancel := context.WithTimeout(ctx, nt.DefaultWaitTimeout)
+	defer timeoutCancel()
+	select {
+	case <-timeoutCtx.Done():
+		nt.T.Fatalf("PortForwarder for %s/%s not ready: %v", ns, deployment, timeoutCtx.Err())
```

Comment:
The error won't include the duration. So you probably want to say something like `Timed out after %v waiting for  PortForwarder for %s/%s to become ready`.

Reply:
Is there a way to get that from the context? Or are you suggesting I use `nt.DefaultWaitTimeout`?

Reply:
done. evaluated using `time`